docs/concept/persistence-storage.md

```diff
@@ -0,0 +1,48 @@
+# Persistence Storage
+Persistence storage is used for unifying data of BanyanDB persistence, including write-ahead logging(WAL), index, and data collected from skywalking and other observability platforms or APM systems. It provides various implementations and IO modes to satisfy the need of different components.
+BanyanDB provides a concise interface that shields the complexity of the implementation from the upper layer. By exposing necessary interfaces, making upper components do not need to care how persistence implement and mask differences between different operating systems.
+
+# IO Mode
+persistence storage support different IO mode to satisfy different throughput requirements. The interface is not directly exposed to developers and can be set through configuration.
+
+## Io_uring
+Io_uring is a new feature in Linux 5.1, which provides high throughput and implements in a fully asynchronous way. In the scene of massive storage, io_uring can bring significant benefits.
+
+## Synchronous IO
+Synchronous IO is the most common IO mode, but the throughput is relatively low. BanyanDB provides a nonblocking mode that can be used in lower Linux versions.
+
+# Operation
+## Directory
+### Create
+Create the specified directory and return an error if the directory already exists.
+
+### Delete
+Delete the directory and all files and return an error if the directory does not exist or the directory not reading or writing.
+
+### rename
+Rename the directory and return an error if the directory already exists.
+
+### Get Children
+Get all lists of files or children's directories in the directory and return an error if the directory does not exist.
+
+## File
+### Write
+BanyanDB provides two methods for writing files.
+Append mode, which adds new data to the end of a file. This mode is typically used for WAL.
+Flush mode, which flushes all data to one or more files and can specify the size of each file to write to.
```

Comment:
When in flush mode, it is important that the file is atomic. This ensures that if an error occurs during flushing, the incomplete file will be discarded.